upgrade/upgrade-provider.go

```diff
@@ -141,23 +143,36 @@ func UpgradeProvider(ctx Context, repoOrg, repoName string) error {
 	if ctx.UpgradeBridgeVersion {
 		discoverSteps = append(discoverSteps,
 			step.F("Planning Bridge Update", func(context.Context) (string, error) {
-				refs, err := gitRefsOf(ctx,
-					"https://github.com/pulumi/pulumi-terraform-bridge.git", "tags")
-				if err != nil {
-					return "", err
-				}
-
-				latest := latestSemverTag("", refs)
+				switch v := ctx.TargetBridgeRef.(type) {
+				case nil:
+					return "", fmt.Errorf("--target-bridge-version is required here")
+				case *HashReference:
+					targetBridgeVersion = v
+				case *Version:
+					// If our target upgrade version is the same as our
+					// current version, we skip the update.
+					if v.String() == goMod.Bridge.Version {
```

Comment:
Should this use `target < current` instead to refuse downgrades?

Reply:
If you manually specify a downgrade, I think it's fine.